### ISSUE TITLE
enter ip address with Apple tv beta integration

### ISSUE THREAD

**Issue by palmerch1** [postlund/pyatv#1706]
### What do you need help with?

Hello,

I am trying to connect my apple tv(tvOS 15.4.1) to HA(v2022.4.5).
I noticed that my version of tvOS wasn't supported by the
"standard" Apple TV integration and discovered the beta version.
So, i installed Apple TV Beta(v3.0.1) via HACS(v1.24.5).
I can see the "Apple TV Beta" integration inside "HACS -> integrations",
but I guess it's not configurated properly as the apple tv is on another
network then my HA instance, but I didn't see any option to connect my apple tv.
My apple tv is in a 192.168.0.0/24 network, while the HA instance is in a 192.168.122.0/24 network.

Can someone please point me towards a config file or some other location where i can enter an ip address ?

I found a file which contains some properties but I don't know if this is the right file : /root/config/.storage/hacs/224041719.hacs

Thank you !

Best regards,
Christof.

**Comment by palmerch1** [comment]
I tried adding "apple_tv:" to configuration.yaml and discovered that there seemed to be an option "- host: ipaddr", but this is not working. And according to the docs I found( https://home-assistant-china.github.io/components/apple_tv/ ), I must also add a login_id, which I should be able to find with "atvremote" which is not available it seems. But when reading the article in more detail, I think this is a rather old article.
But on the other hand atvremote seems like a python script that is still being developed : https://github.com/postlund/pyatv/tree/master/pyatv/scripts

Can someone please guide me in the right direction on how to configure my apple tv(tvOS 15.4.1) with home assistant ?

Thank you !

Best regards,
Christof.

**Comment by postlund** [comment]
That documentation is really old and not valid anymore. You can only add new devices via UI, I.e. yaml is not supported. If the device you are adding is on another network, you *must* set up MDNS relaying (I.e. something that forwards the MDNS messages feel your Apple TV to the VLAN that home assistant is on). There's no way around that. Also, you don't need the beta version. Just run with the one bundled with Home Assistant, it's more up to date anyway right now.

**Comment by postlund** [comment]
Did you manage to get it working?

**Comment by palmerch1** [comment]
Hello Pierre,

Sorry for the late reply. I still have to change the HA hypervisor setup in order to get the MDNS traffic to pass between the HA VM and the apple tv device.

I found this URL which i think/hope will fix the issue to let the MDNS traffic through:
https://levelup.gitconnected.com/how-to-setup-bridge-networking-with-kvm-on-ubuntu-20-04-9c560b3e3991

Best regards,
Christof.

**Comment by palmerch1** [comment]
Hello Pierre,

Finally I was able to change my KVM-qemu network configuration from the default NAT type bridge to a bridge on which the HA VM is on the same network as the hypervisor and using the same DHCP server.

By doing this, HA suddenly indicated that it had discovered a lot of new devices(I have never seen HA doing this) and one of these devices was the apple tv :-)
When I clicked the configure button for the apple tv integration, it guided me through the process by asking me for a number of PIN codes that were displayed on the TV connected to my apple TV. Now it seems to be setup correctly.

So the issue was the networking mode of KVM-qemu. The link in my previous post provided me with enough info to change my network setup.

Thanks again for the support !!

Best regards,
Christof.
